includes/class-wcsr-resource.php

```diff
@@ -187,24 +188,96 @@ public function get_days_active( $from_timestamp, $to_timestamp = null ) {
+			$start_timestamp = ( $this->get_date_created()->getTimestamp() > $from_timestamp ) ? $this->get_date_created()->getTimestamp() : $from_timestamp;
+			array_unshift( $activation_times, $start_timestamp );
 		}
 
 		foreach ( $activation_times as $i => $activation_time ) {
-			if ( isset( $deactivation_times[ $i ] ) ) {
-				$days_active -= floor( ( $activation_time - $deactivation_times[ $i ] ) / DAY_IN_SECONDS );
+			// If there is corresponding deactivation timestamp, the resouce has deactivated before the end of the period so that's the time we want, otherwise, use the end of the period as the resource was still active at end of the period
+			$deactivation_time = isset( $deactivation_times[ $i ] ) ? $deactivation_times[ $i ] : $to_timestamp;
+
+			// skip over any days that are activated/deactivated on the same day and have already been accounted for
+			if ( $i !== 0 && self::is_on_same_day( $deactivation_time, $deactivation_times[ $i - 1 ], $from_timestamp ) ) {
+				continue;
+			}
+
+			// Calculate days based on time between
+			$days_by_time = intval( ceil( ( $deactivation_time - $activation_time ) / DAY_IN_SECONDS ) );
+
+			// Increase our tally
+			$days_active += $days_by_time;
+
+			// If days based on time is only 1 but it was "across a day" we may need to adjust IF NOT accounted for already
+			if ( $days_by_time == 1 && ! self::is_on_same_day( $activation_time, $deactivation_time, $from_timestamp ) ) {
+
+				// handle situation if first activation crosses a day
+				if ( $i == 0 && ! self::is_on_same_day( $activation_time, $deactivation_time, $from_timestamp ) ) {
+					$days_active += 1;
+				}
+
+				// if this activation didn't start on the same day as previous activation it is safe to add an extra day
+				if ( $i !== 0 && ! self::is_on_same_day( $activation_time, $deactivation_times[ $i - 1 ], $from_timestamp ) ) {
+					$days_active += 1;
+				}
 			}
 		}
 
 		return $days_active;
 	}
 
 	/**
+	 * Prototype/demo conditional check for whether a timestamp is on the same "day" as another timestamp
+	 *
+	 * The catch is the "day" is not typical calendar day - it based on a 24 hour period from initial activation
+	 *
+	 * Initial activation could be the actual time it was first activated of the start of the period.
+	 *
+	 * Takes the activation/starting timestamp and gets the time our days start from.
+	 * Takes the timestamp we are checking if we are on the same day as and gets the date, previous days date and time
+	 * Works out whether the current day started on the same date as the one we are comparing with or a day earlier and works makes a time stampe fo when the day starts
+	 * Works out when the days ends
+	 * Check if our timestamp is with the start and end dates we have determined
+	 *
+	 * @param  int  $current_timestamp    [description]
+	 * @param  int  $compare_timestamp [description]
+	 * @param  int  $start_timestamp  [description]
+	 * @return boolean true on same day | false if not
```

Comment:
GH is marking this PR as not succeeding checks because this line wasn't specifically changed. But i updated the doc block to remove the "activation" usage in ce3a57f :)